libs/usb-drive/src/usb_drive.ts

```diff
@@ -0,0 +1,143 @@
+import { promises as fs } from 'fs';
+import { join } from 'path';
+import { execFile } from 'child_process';
+import { promisify } from 'util';
+import { assertDefined } from '@votingworks/basics';
+import makeDebug from 'debug';
+
+const exec = promisify(execFile);
+
+const debug = makeDebug('usb-drive');
+
+export type UsbDriveStatus =
+  | { status: 'no_drive' }
+  | { status: 'mounted'; mountPoint: string }
+  | { status: 'ejected' }
+  | { status: 'error'; reason: 'bad_format' };
+
+export interface UsbDrive {
+  status(): Promise<UsbDriveStatus>;
+  eject(): Promise<void>;
+  format(): Promise<void>;
+}
+
+interface BlockDeviceInfo {
+  name: string;
+  path: string;
+  mountpoint: string | null;
+  fstype: string | null;
+  fsver: string | null;
+  label: string | null;
+}
+
+async function getBlockDeviceInfo(
+  devicePath: string
+): Promise<BlockDeviceInfo | undefined> {
+  try {
+    const { stdout } = await exec('lsblk', [
+      '-J',
+      '-n',
+      '-l',
+      '-o',
+      ['NAME', 'MOUNTPOINT', 'FSTYPE', 'FSVER', 'LABEL'].join(','),
+      devicePath,
+    ]);
+    const rawData = JSON.parse(stdout) as {
+      blockdevices: BlockDeviceInfo[];
+    };
+    debug(`Got block device info for ${devicePath}: ${stdout}`);
+    return { ...assertDefined(rawData.blockdevices[0]), path: devicePath };
+  } catch (error) {
+    debug(`Error getting block device info for ${devicePath}: ${error}`);
+    return undefined;
+  }
+}
+
+async function findUsbDriveDevice(): Promise<string | undefined> {
+  const DEVICE_PATH_PREFIX = '/dev/disk/by-id/';
+  const USB_REGEXP = /^usb(.+)part(.*)$/;
+
+  // List devices, filtering to only the USB partitions
+  const devicesById = (await fs.readdir(DEVICE_PATH_PREFIX)).filter((name) =>
+    USB_REGEXP.test(name)
+  );
+
+  // We only support one USB drive at a time
+  const [deviceId] = devicesById;
+  if (!deviceId) {
+    return undefined;
+  }
+
+  // Follow the symlink
+  const devicePath = join(
+    DEVICE_PATH_PREFIX,
+    await fs.readlink(join(DEVICE_PATH_PREFIX, deviceId))
+  );
+
+  return devicePath;
+}
+
+async function getUsbDriveStatus(): Promise<BlockDeviceInfo | undefined> {
+  const devicePath = await findUsbDriveDevice();
+  if (!devicePath) return undefined;
+  debug(`Found USB drive at ${devicePath}`);
+  return await getBlockDeviceInfo(devicePath);
+}
+
+const VX_MOUNT_POINT = '/media/vx/usb-drive';
+
+async function mountUsbDrive(devicePath: string): Promise<void> {
+  debug(`Mounting USB drive ${devicePath} at ${VX_MOUNT_POINT}`);
+  await fs.mkdir(VX_MOUNT_POINT, { recursive: true });
+  await exec('mount', [
+    '-w',
+    '-o',
+    'umask=000,nosuid,nodev,noexec',
+    devicePath,
+    VX_MOUNT_POINT,
+  ]);
```

Comment:
Ok to call these commands directly instead of using the `mount-usb.sh` app scripts from vxsuite-complete-system?

Reply:
We have a preference for using scripts, see @benadida's comment [here](https://github.com/votingworks/kiosk-browser/pull/142#issuecomment-1382585164). I think this case is analogous but not confident.

To make the above work, would you not not need to call `mount` with `sudo`? And to call `mount` with `sudo`, you'd have to give the running user complete, no-password access to `sudo mount` via `sudoers.etc` (there are ways to restrict what arguments a command can be called with in the `sudoers.etc` file, but not a way to allow a dynamic argument like the device path we'll be passing in). We'd prefer to only give the running user access to a script so, if compromised, they could not mount whatever they like.

Reply:
Thanks for the context, I figured there was a reason.

Should these scripts continue to live in vxsuite-complete-system? If so, how do we support accessing them in dev? Maybe we have a dev version of the scripts in vxsuite?

Reply:
I'm not sure. 

The way `kiosk-browser` supports it in dev is just assuming there's a `vxsuite-complete-system` repo adjacent to a `kiosk-browser` repo, which could be finicky but we've had no problems. We'd still have to figure out and test pathing so that the scripts are accessible when locked down.

I think it would probably be better to move them into `vxsuite` to make pathing to it intuitive and have all the code colocated. There are a couple consumers in `vxsuite-complete-system` but they could use a path to `vxsuite` or copies of the scripts. I'd need to refresh my memory on what happens to `vxsuite` code when we lock down, and how to path to it (from `sudoers.etc`).

Reply:
Colocating the USB mount/unmount scripts in vxsuite would definitely make development easier. If that's workable for prod, I'd support that.

@amcmanus Do you have any thoughts on this?

Reply:
I think putting the scripts in vxsuite makes sense. If there's system level config that needs to happen in support of script functionality, that'll be handled by complete-system / build-system.

Reply:
Just thought I'd note that, while working on auth changes, I also found it really useful to be able to put scripts in vxsuite (instead of vxsuite-complete-system). And I did in fact end up having a vxsuite-complete-system script consume a vxsuite script ([link](https://github.com/votingworks/vxsuite-complete-system/blob/c77994a22449046ddcb1d0e18dcafe7112850215/config/admin-functions/program-system-administrator-cards.sh#L12-L20))

Reply:
Updated to use scripts here: a39278fea

Reply:
> > how do you think this would be mocked in an integration-testing environment?
> 
> I haven't worked with the integration tests much. Do you think using the provided `MockUsbDrive` would work?

I don't think so because, although we could have a flag to use `MockUsbDrive`, we wouldn't be able to access its API.   We need an exposed API. We could do a [`MockFileUsbDrive`](https://github.com/votingworks/vxsuite/blob/main/libs/auth/src/mock_file_card.ts) similar to what @arsalansufi made for `libs/auth`.

Reply:
@amcmanus can you point me to what I might need to modify in complete-system/build-system to support these new scripts I'm adding?